README.md

```diff
@@ -1,45 +1,191 @@
+
+It supports helm chart installations from two sources: **_helm repositories_** and **_local paths_**. Additionally, it helps to process additional local installations of CRs, CRDs and custom checks.
+The library could be used to simply process deployments on target clusters or use withing your own operator to carry deployment operations. 
+E.g. [template-operator](https://github.com/kyma-project/lifecycle-manager/tree/main/samples/template-operator) used this library (via the [declarative](operator/pkg/declarative) library) to perform necessary operations on target clusters during reconciliations.
+Import package `github.com/kyma-project/module-manager/operator/pkg/manifest` to include the main functionality required to process helm charts coupled with additional state handling.
+For more options and information refer to the [InstallInfo](operator/pkg/manifest/operations.go) type definition.
+
+Sample usage:
+
+```go
+package sample
+// Sample usage of chart installation via local chart path
+
+import (
+    "k8s.io/client-go/rest"
+    "sigs.k8s.io/controller-runtime/pkg/client"
+    "github.com/kyma-project/module-manager/operator/pkg/manifest"
+	"github.com/kyma-project/module-manager/operator/pkg/types"
+	"github.com/kyma-project/module-manager/operator/pkg/custom"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var restConfig *rest.Config
+var client client.Client
+
+deployInfo := manifest.InstallInfo{
+    Ctx: ctx,
+    ChartInfo: &manifest.ChartInfo{
+        ChartPath:   "/chart/path",
+        Flags:       types.ChartFlags{
+            // ConfigFlags support string, bool and int types as helm chart flags
+            // check: https://github.com/helm/helm/blob/d7b4c38c42cb0b77f1bcebf9bb4ae7695a10da0b/pkg/action/install.go#L67
+            ConfigFlags: types.Flags{ 
+                "Namespace":       chartNs,
+                "CreateNamespace": true,
+            },
+            // SetFlags are chart value overrides
+            SetFlags: types.Flags{
+                ".some.value.override": "override",
+            },      
+        },
+    },
+    ClusterInfo: custom.ClusterInfo{
+        Config: restConfig, // destination cluster rest config
+        Client: client, // destination cluster rest client
+    },
+    ResourceInfo: manifest.ResourceInfo{
+        CustomResources: []*unstructured.Unstructured{} // additional custom resources to be installed
+        BaseResource: unstructured.Unstructured{}, // base resource to be passed for custom checks, usually the reconciled resource
+		Crds: []*apiextensions.CustomResourceDefinition // additional custom resource definitions to be installed
+    },
+    CheckReadyStates: true,
+}
+
+ready, err := manifest.InstallChart(logger, deployInfo, []types.ObjectTransform{})
+if err != nil {
+	return false, err
+}
+
+ready, err := manifest.UninstallChart(logger, deployInfo, []types.ObjectTransform{})
```

Comment:
why is this uninstall immediately running after install?
I think this sample should be working ootb, not sure if it is doing that right now

Reply:
Good point, I have moved `ConsistencyCheck` before `Uninstall`. The idea is to specify all 3 options. I have added additional comments now.

Reply:
Hmm, Im still a bit uncertain about the choice of representation there. Why don't we implement a full reconciliation here that handles all 3 like we do in template operator? Is it too big?

Reply:
IMO that is unecessary since `deployInfo` will remain the same. Having `deployInfo` 3 times will be an overkill.

Reply:
I think its rather about the reconciliation being complete in the example, otherwise it wont be able to deal with uninstallation/upgrades